### PR TITLE
feat: Server-Side Encryption for SQS Queues

### PR DESCRIPTION
**Pull request by zucchinidev**
[#187061621](https://www.pivotaltracker.com/story/show/187061621)

### Checklist:

* [ ] Have you added Release Notes in the docs repositories?
* [x] Have you ran `make run-integration-tests` and `make run-terraform-tests`?
* [x] Have you ran acceptance tests for the service under change?
* [x] Have you followed the [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary)?

### CODE REVIEW

**Review by pivotal-marcela-campo — APPROVED** [review]

File: integration-tests/integration_test_suite_test.go [hunk 1]
```diff
@@ -53,6 +53,7 @@ var _ = BeforeSuite(func() {
 		"AWS_SECRET_ACCESS_KEY=" + awsSecretAccessKey,
 		"CSB_LISTENER_HOST=localhost",
 		"GSB_COMPATIBILITY_ENABLE_BETA_SERVICES=true",
+		"TERRAFORM_UPGRADES_ENABLED=false",
```
Comment (FelisiaM) [inline integration-tests/integration_test_suite_test.go:56]: was this intentional and is it needed?

File: terraform/sqs/bind/data.tf [hunk 2]
```diff
@@ -1,15 +1,7 @@
...
-      sid       = try(statement.value.sid, null)
-      actions   = try(statement.value.actions, null)
-      resources = try(statement.value.resources, null)
-    }
-  }
-}
-
 locals {
+  key_ids_list = compact(split(",", var.kms_all_key_ids))
+  has_key_ids  = length(local.key_ids_list) != 0
```
Comment (FelisiaM) [inline terraform/sqs/bind/data.tf:3]: nitp: Declering these closer to where they are used makes it easier to read navigate this file.

**Review by FelisiaM — COMMENTED** [review]
Overall it looks good! I've left a couple of comments to tidy up

File: integration-tests/integration_test_suite_test.go (hunk 1 above)
Comment (zucchinidev) [reply integration-tests/integration_test_suite_test.go:56]: Good catch! thanks!

File: terraform/sqs/provision/main.tf [hunk 3]
```diff
@@ -15,6 +15,15 @@ resource "aws_sqs_queue" "queue" {
   deduplication_scope   = var.deduplication_scope
   fifo_throughput_limit = var.fifo_throughput_limit
 
+  # Server-side encryption settings
+  kms_master_key_id                 = var.kms_master_key_id == "" ? null : var.kms_master_key_id
+  kms_data_key_reuse_period_seconds = var.kms_data_key_reuse_period_seconds
+
+  # Conflicting configuration arguments: `sqs_managed_sse_enabled` conflicts with `kms_master_key_id`
+  # We could use nullable boolean type, but the UX would not be ideal:
+  # E.g: {"kms_master_key_id": "alias/aws/sqs", "sqs_managed_sse_enabled": null}
```
[the remark this answers is not in the document]
Comment (zucchinidev) [reply terraform/sqs/provision/main.tf:24]: I'll delete it. Thank you very much

File: terraform/sqs/bind/data.tf (hunk 2 above)
Comment (zucchinidev) [reply terraform/sqs/bind/data.tf:3]: Sounds good to me.

File: scripts/push-broker.sh [hunk 4]
```diff
@@ -139,4 +139,4 @@ if [[ -z ${BROKER_NAME} ]]; then
   BROKER_NAME=csb-$USER
 fi
 
-cf create-service-broker "${BROKER_NAME}" "${SECURITY_USER_NAME}" "${SECURITY_USER_PASSWORD}" https://$(LANG=EN cf app "${APP_NAME}" | grep 'routes:' | cut -d ':' -f 2 | xargs) --space-scoped --update-if-exists
+cf create-service-broker "${BROKER_NAME}" "${SECURITY_USER_NAME}" "${SECURITY_USER_PASSWORD}" https://$(LANG=EN cf app "${APP_NAME}" | grep 'routes:' | cut -d ':' -f 2 | xargs) --space-scoped || cf update-service-broker "${BROKER_NAME}" "${SECURITY_USER_NAME}" "${SECURITY_USER_PASSWORD}" https://$(LANG=EN cf app "${APP_NAME}" | grep 'routes:' | cut -d ':' -f 2 | xargs)
```
[the remark this answers is not in the document]
Comment (zucchinidev) [reply scripts/push-broker.sh]: Done! Thank you very much :heart:

**Review by FelisiaM — APPROVED** [review]

**Pull request merged**